src/client/address_book/storage.cc

```diff
@@ -68,15 +69,20 @@ bool AddressBookStorage::GetAddress(
 void AddressBookStorage::AddAddress(
     const kovri::core::IdentityEx& address) {
   auto filename = GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
-  std::ofstream file(filename.string(), std::ofstream::binary);
-  if (!file)
+  kovri::core::OutputFileStream file(filename.string(), std::ofstream::binary);
+  if (file.Bad() || file.Fail())
     LOG(error) << "AddressBookStorage: can't open file " << filename;
   const std::size_t len = address.GetFullLen();
   auto buf = std::make_unique<std::uint8_t[]>(len);
   // For sanity, the validity of identity length is incumbent upon the parent caller.
   // TODO(unassigned): triple check that this is the case
   address.ToBuffer(buf.get(), len);
-  file.write(reinterpret_cast<char *>(buf.get()), len);
+  if (!file.Write(buf.get(), len))
```

Comment:
>const std::size_t len = address.GetFullLen();
>auto buf = std::make_unique<std::uint8_t[]>(len);
>address.ToBuffer(buf.get(), len);
>if (!file.Write(buf.get(), len))

While you're here (though it's not mandatory for this PR), since this is a refactor PR, you can refactor the buffer into a vector so then you'll only need to use `buf.data()` and `buf.size()`.